managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -4321,4 +4321,44 @@ public void testDeleteCurrentLedgerWhenItIsClosed(boolean closeLedgerByAddEntry)
             assertEquals(ml.currentLedgerEntries, 0);
         });
     }
+
+    @Test
+    public void testReverseFindPositionOneByOne() throws Exception {
+        final int maxEntriesPerLedger = 5;
+
+        ManagedLedgerConfig managedLedgerConfig = new ManagedLedgerConfig();
+        managedLedgerConfig.setMaxEntriesPerLedger(maxEntriesPerLedger);
+        ManagedLedger ledger = factory.open("testReverseFindPositionOneByOne", managedLedgerConfig);
+        
+        String matchEntry = "match-entry";
+        String noMatchEntry = "nomatch-entry";
+        Predicate<Entry> predicate = entry -> {
+            try {
+                String entryValue = entry.getDataBuffer().toString(UTF_8);
+                if (matchEntry.equals(entryValue)) {
+                    return true;
+                }
+            } finally {
+                entry.release();
+            }
```

Comment:
it's better to move releasing to internalAsyncReverseFindPositionOneByOne ?

Reply:
Make sense, moved.